suc/update-connection-info.ps1

```diff
@@ -0,0 +1,172 @@
+                $env:RANCHER_SUCCESS = $true
+                $retries = 99
+                break
+            }
+            default {
+                Write-LogError "$responseCode received while testing Rancher connection. Sleeping for 5 seconds and trying again."
+                Start-Sleep -Seconds 5
+                $retries++
+                continue
+            }
+        }
+    }
+}
+
+function Test-CaRequired {
+    $env:CA_REQUIRED = $false
+    if ($env:CATTLE_SERVER) {
+        $retries = 0
+        while ($retries -lt 6) {
+            curl.exe --connect-timeout 60 --max-time 60 -sfL "$env:CATTLE_SERVER/healthz"
+            Write-LogDebug "Received curl exit code $EXITCODE"
+            switch ($LASTEXITCODE) {
+                0 {
+                    Write-LogInfo "Determined CA is not necessary to connect to Rancher."
+                    $env:CATTLE_CA_CHECKSUM = ""
+                    $retries = 99
+                    break
+                }
+                { $_ -in 60, 77, 35 } {
+                    Write-LogInfo "Determined CA is necessary to connect to Rancher."
+                    $env:CA_REQUIRED = $true
+                    $retries = 99
+                    break
+                }
+                default {
+                    Write-LogError "Error while connecting to Rancher to verify CA necessity. Sleeping for 5 seconds and trying again. Received error code $LASTEXITCODE"
+                    Start-Sleep -Seconds 5
+                    $retries++
+                    continue
+                }
+            }
+        }
+    } else {
+        Write-LogWarn "`$env:CATTLE_SERVER was not provided, cannot determine if a CA is required"
+    }
+}
+
+function Get-RancherConnectionInfo() {
+    $retries = 0
+    Write-LogInfo "Attempting to get updated Rancher connection info"
+    while ($retries -lt 6) {
+        $responseCode = $(curl.exe --connect-timeout 60 --max-time 60 --write-out "%{http_code}\n " --ssl-no-revoke -sfL "$env:CATTLE_SERVER/v3/connect/agent" -o $env:CATTLE_AGENT_VAR_DIR/rancher2_connection_info.json -H "Authorization: Bearer $($env:CATTLE_TOKEN)" -H "X-Cattle-Id: $($env:CATTLE_ID)" -H "Content-Type: application/json")
+        switch ( $responseCode ) {
+            { $_ -in "ok200", 200 } {
+                Write-LogInfo "Successfully downloaded Rancher connection information."
+                $retries = 99
+                break
+            }
+            default {
+                Write-LogError "$responseCode received while downloading Rancher connection information. Sleeping for 5 seconds and trying again."
+                Start-Sleep -Seconds 5
+                $retries++
+                continue
+            }
+        }
+    }
+}
+
+function Update-ConnectionInfo()
+{
+    $env:RKE2_DATA_DIR = "c:\var\lib\rancher"
+    $env:CATTLE_AGENT_CONFIG_DIR = "c:\etc\rancher\wins"
+
+    $env:CATTLE_AGENT_VAR_DIR = "$env:RKE2_DATA_DIR\agent"
```

Comment:
I'm a bit worried that, due to `RKE2_DATA_DIR` / `CATTLE_AGENT_VAR_DIR` not being passed properly to the system-agent-installer on initial provisioning, using it properly here might result in us using the wrong location. The intent with this PR is to isolate it to the 0.4.x release line, which will be limited to Rancher 2.9 and below. Because we don't have data dir support for windows on 2.9.x (and I'm not sure we plan on fixing that, but you can correct me if I'm wrong) I don't know if we should consider the value when updating the connection info script.

Reply:
Which variable specifically?